integrations/third_party/wiz/tests/core/session.py

```diff
@@ -0,0 +1,121 @@
+        invalid_response = handle_invalid_response(issue.issue_id)
+        if invalid_response is not None:
+            return invalid_response
+
+        return MockResponse(
+            content={"data": {"issue": issue.to_json()}},
+            status_code=200,
+        )
+
+    def _add_comment_to_issue(self) -> MockResponse:
+        """Handle add comment to issue request's response."""
+        issue: datamodels.Issue = self._product.get_issues()[0]
+        invalid_response: MockResponse = handle_invalid_response(issue_id=issue.issue_id)
+        if invalid_response is not None:
+            return invalid_response
+        comment: common.Comment = self._product.get_comment(issue.issue_id)
+        comment.is_comment = True
+
+        return MockResponse(content=common.ADD_COMMENT_TO_ISSUE, status_code=200)
+
+    def _reopen_issue(self) -> MockResponse:
+        """Handle reopen issue request's response."""
+        issue: common.IssueStatus = self._product.get_issues()[0]
+        invalid_response: MockResponse = handle_invalid_response(issue_id=issue.issue_id)
+        if invalid_response is not None:
+            return invalid_response
+        issue.status = constants.STATUS_REOPEN
+
+        return MockResponse(content=common.REOPEN_ISSUE, status_code=200)
+
+    def _ignore_issue(self) -> MockResponse:
+        """Handle reopen issue request's response."""
+        issue: common.IssueStatus = self._product.get_issues()[0]
+        invalid_response: MockResponse = handle_invalid_response(issue_id=issue.issue_id)
+        if invalid_response is not None:
+            return invalid_response
+        issue.status = constants.STATUS_REJECTED
+
+        return MockResponse(content=common.IGNORE_ISSUE, status_code=200)
+
+    def _resolve_issue(self) -> MockResponse:
+        """Handle reopen issue request's response."""
+        issue: common.IssueStatus = self._product.get_issues()[0]
+        invalid_response: MockResponse = handle_invalid_response(issue_id=issue.issue_id)
+        if invalid_response is not None:
+            return invalid_response
+        issue.status = constants.STATUS_RESOLVED
+
+        return MockResponse(content=common.RESOLVE_ISSUE, status_code=200)
+
+
+@router.post("/oauth/token")
+def get_oauth_token(request: MockRequest) -> MockResponse:
+    """Get an OAuth token"""
+    if "raise_error" in request.kwargs["data"].values():
+        return MockResponse(
+            content=common.INVALID_TOKEN,
+            status_code=400,
+        )
+
+    return MockResponse(common.VALID_TOKEN, status_code=200)
+
+
+def handle_invalid_response(issue_id) -> MockResponse | None:
+    """Handle invalid response."""
+    if issue_id == str(common.INVALID_ISSUE_ID):
+        return MockResponse(content=common.INVALID_ISSUE, status_code=200)
+    if issue_id == common.INVALID_TOKEN_ID:
+        return MockResponse(content=common.INVALID_TOKEN, status_code=401)
+
+    return None
```

Comment:
For more simplicity you can use match case here to match the issue_id.

```suggestion
    """Handle invalid response."""
    match issue_id:
        case str(common.INVALID_ISSUE_ID):
                return MockResponse(content=common.INVALID_ISSUE, status_code=200)
        case common.INVALID_TOKEN_ID:
                return MockResponse(content=common.INVALID_TOKEN, status_code=401)

    return None
```